// components/pages/signin/index.js
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import i18next from "i18next";

class Signin extends Component {

    constructor(props) {
        super(props);
        this.handleEmailChange = this.handleEmailChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
        this.state = {
            email: '',
            password: ''
        };
    }

    handleEmailChange(e) {
        this.setState({email: e.target.value})
    }

    handlePasswordChange(e) {
        this.setState({password: e.target.value})
    }

    signIn() {
        alert('Email address is ' + this.state.email + ' Password is ' + this.state.password);
    }

    render() {
        return (
            <div className={"sign"}>
                <form className="form-signin">
                    <h2 className="form-signin-heading">
                        {i18next.t("Please sign in")}
                    </h2>
                    <label htmlFor="inputEmail" className="sr-only">
                        Email address
                    </label>
                    <input
                        onChange={this.handleEmailChange}
                        type="email"
                        id="inputEmail"
                        className="form-control"
                        placeholder={i18next.t("Email address")}
                        required
                        autoFocus
                    />
                    <label
                        htmlFor="inputPassword"
                        className="sr-only"
                    >
                        Password
                    </label>
                    <input
                        onChange={this.handlePasswordChange}
                        type="password"
                        id="inputPassword"
                        className="form-control"
                        placeholder={i18next.t("Password")}
                        required/>
                    <button
                        className="btn btn-lg btn-primary btn-block"
                        onClick={this.signIn.bind(this)}
                        type="button"
                    >
                        {i18next.t("Sign in")}
                    </button>
                    <div>
                        {/*TODO: Remember me*/}
                        {/*<input*/}
                            {/*name="isGoing"*/}
                            {/*type="checkbox"*/}
                            {/*checked={this.state.isGoing}*/}
                            {/*onChange={this.handleInputChange}*/}
                        {/*/>*/}
                        {/*Remember me*/}
                    </div>
                    <div>
                        <Link to='/signup'> {i18next.t("Registration")} </Link>
                    </div>
                    {/*TODO: Forgot password?*/}
                </form>
            </div>
        );
    }
}

export default Signin;
