// containers/menu/index.js
import React, {Component} from "react";
import SideNav, {Nav, NavIcon, NavText} from 'react-sidenav';
import {Link} from 'react-router-dom';
import i18next from "i18next";

// Icons
import SvgIcon from 'react-icons-kit';
import {ic_aspect_ratio} from 'react-icons-kit/md/ic_aspect_ratio';
import {ic_business} from 'react-icons-kit/md/ic_business';
import {home} from 'react-icons-kit/icomoon/home';
import {menu} from 'react-icons-kit/icomoon/menu';
import {sun} from 'react-icons-kit/icomoon/sun';
import {cogs} from 'react-icons-kit/icomoon/cogs';
import {user} from 'react-icons-kit/icomoon/user';
import {tv} from 'react-icons-kit/icomoon/tv';

import myImg from 'images/vovaks.png';

class Menu extends Component {

    constructor(props) {
        super(props);
        this.state = {
            lng: 'en',
        };
    }

    componentWillMount() {
        this.setLanguage(this.state.lng);
    }

    /**
     * setLanguage
     * @param language
     */
    setLanguage(language) {
        console.log({language});
        i18next.init({
            lng: language,
            resources: {
                "ru": {
                    "translation":
                        require('translate/ru.json')
                },
                "en": {
                    "translation":
                        require('translate/en.json')
                },
            },
        });
    }


    /**
     * OnChange for Language Selector
     * @param event
     */
    handleChangeLanguage = (event) => {
        this.setState({lng: event.target.value});
        this.setLanguage(event.target.value);
    };

    render() {
        return (
            <div style={{width: 200, color: '#FFF'}}>
                <div style={{display: 'flex', padding: 16, background: '#1a2229'}}>
                    <div style={{width: 40, height: 40}}>
                        <img
                            src={myImg}
                            alt="User"
                            style={{borderRadius: '30px', width: 40, height: 40}}
                        />
                    </div>
                    <div style={{paddingLeft: 6, paddingTop: 6}}>
                        <div style={{fontSize: 12, color: '#E5E5E5'}}>
                            {' '}Vovaks{' '}
                        </div>
                        <div style={{fontSize: 11}}> Developer</div>
                    </div>
                </div>
                <select
                    id="lang"
                    onChange={this.handleChangeLanguage}
                    value={this.state.value}
                >
                    <option value='en'>English</option>
                    <option value='ru'>Russian</option>
                </select>
                {/*<SideNav highlightColor='#E91E63' highlightBgColor='#00bcd4' defaultSelected='menu'>*/}
                <SideNav
                    highlightColor='#222'
                    highlightBgColor='#00bcd4'
                    defaultSelected='home'
                >
                    <Nav id='menu'>
                        <NavIcon><SvgIcon size={20} icon={menu}/></NavIcon>
                        <NavText> {i18next.t('Menu')}</NavText></Nav>
                    <Link to='/'>
                        <Nav id='home'>
                            <NavIcon><SvgIcon size={20} icon={home}/></NavIcon>
                            <NavText> {i18next.t('Index')} </NavText>
                        </Nav></Link>
                    <Link to='/tv'>
                        <Nav id='tv'>
                            <NavIcon><SvgIcon size={20} icon={tv}/></NavIcon>
                            <NavText> {i18next.t('Tv')} </NavText>
                        </Nav></Link>
                    <Link to='/dashboard'>
                        <Nav id='dashboard'>
                            <NavIcon><SvgIcon size={20} icon={ic_aspect_ratio}/></NavIcon>
                            <NavText> {i18next.t('Board')} </NavText>
                        </Nav></Link>
                    <Link to='/sales'>
                        <Nav id='sales'>
                            <NavIcon><SvgIcon size={20} icon={ic_business}/></NavIcon>
                            <NavText> {i18next.t('Sales')} </NavText>
                        </Nav>
                    </Link>
                    <Link to='/weather'>
                        <Nav id='sun'>
                            <NavIcon><SvgIcon size={20} icon={sun}/></NavIcon>
                            <NavText> {i18next.t('Weather')} </NavText>
                        </Nav></Link>

                    <Link to='/cogs'>
                        <Nav id='cogs'>
                            <NavIcon><SvgIcon size={20} icon={cogs}/></NavIcon>
                            <NavText> {i18next.t('Configurate')} </NavText>
                        </Nav></Link>

                    <Link to='/user'>
                        <Nav id='user'>
                            <NavIcon><SvgIcon size={20} icon={user}/></NavIcon>
                            <NavText>  {i18next.t('User')}</NavText>
                        </Nav></Link>
                </SideNav>
            </div>
        );
    }
}

export default Menu;
